aemedge/blocks/hero/hero.js

```diff
@@ -78,6 +78,9 @@ export default async function decorate(block) {
   // get images for background
   let picture = block.querySelector(':scope div > div > picture');
   if (picture) {
+    if (block.classList.contains('pattern')) {
+      picture.innerHTML = picture.innerHTML.replaceAll('webply', 'webpll');
```

Comment:
Should we only replace it for the right src attributes or the picture & img tags?

Reply:
I think we should either enhance createOptimizedPicture to accept a format hint between the two and then replace the picture with the result of createOptimizedPicture call or at a minimum, we should iterate over the source elements and only replace the scrset if the type is image/webpl

Reply:
> Should we only replace it for the right src attributes or the picture & img tags?

I had a look at the server-side generated block, and this is what I got:
<img width="1346" alt="image" src="https://github.com/urfuwo/hlx-test/assets/33486726/98ec6cd1-0ce2-4b98-8cea-1c6cfa890936">

I believe the relevant part is not the source type="...", but rather the "format=webply" parameter in the URL. As you can see, there's also other entries with "format=png" that I don't (want to) touch.

I therefore would suggest to replace "format=webply" with "format=webpll", as the decision to generate the target format "webply" (= lossy webp) is made by the EDS server-side code, and we only want to correct that

Reply:
> I think we should either enhance createOptimizedPicture to accept a format hint between the two and then replace the picture with the result of createOptimizedPicture call or at a minimum, we should iterate over the source elements and only replace the scrset if the type is image/webpl

I don't fully understand the approach. The <picture> element is generated server-side, so our JS code (createOptimizedPicture) does not generate it.

Can you please detail your approach, @karlpauls ?